src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -199,41 +200,62 @@ public static void RangePass()
+                Assert.Equal(float.NegativeInfinity, JsonSerializer.Parse<float?>(float.MinValue.ToString(CultureInfo.InvariantCulture) + "0").Value);
+                Assert.Equal(float.PositiveInfinity, JsonSerializer.Parse<float?>(float.MaxValue.ToString(CultureInfo.InvariantCulture) + "0").Value);
+
+                Assert.Equal(double.NegativeInfinity, JsonSerializer.Parse<double>(double.MinValue.ToString(CultureInfo.InvariantCulture) + "0"));
+                Assert.Equal(double.PositiveInfinity, JsonSerializer.Parse<double>(double.MaxValue.ToString(CultureInfo.InvariantCulture) + "0"));
+                Assert.Equal(double.NegativeInfinity, JsonSerializer.Parse<double?>(double.MinValue.ToString(CultureInfo.InvariantCulture) + "0").Value);
+                Assert.Equal(double.PositiveInfinity, JsonSerializer.Parse<double?>(double.MaxValue.ToString(CultureInfo.InvariantCulture) + "0").Value);
+            }
 
             // Verify sign is correct.
-            // On NETFX a value of -0 does not keep the sign.
-            Assert.Equal(0x0000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("0")));
-            Assert.Equal(0x8000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("-0")));
-            Assert.Equal(0x8000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("-0.0")));
-
-#if BUILDING_INBOX_LIBRARY
-            // Verify sign is correct; SingleToInt32Bits not available on netfx.
-            Assert.Equal(0x00000000u, (uint)BitConverter.SingleToInt32Bits(JsonSerializer.Parse<float>("0")));
-            Assert.Equal(0x80000000u, (uint)BitConverter.SingleToInt32Bits(JsonSerializer.Parse<float>("-0")));
-            Assert.Equal(0x80000000u, (uint)BitConverter.SingleToInt32Bits(JsonSerializer.Parse<float>("-0.0")));
-#endif
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Equal(0x00000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("0")));
+                Assert.Equal(0x00000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("-0")));
+                Assert.Equal(0x00000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("-0.0")));
+
+                Assert.Equal(0x0000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("0")));
+                Assert.Equal(0x0000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("-0")));
+                Assert.Equal(0x0000000000000000ul, (ulong)BitConverter.DoubleToInt64Bits(JsonSerializer.Parse<double>("-0.0")));
+            }
+            else
+            {
+                Assert.Equal(0x00000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("0")));
+                Assert.Equal(0x80000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("-0")));
+                Assert.Equal(0x80000000u, (uint)SingleToInt32Bits(JsonSerializer.Parse<float>("-0.0")));
```

Comment:
Since 0x8 and 0x0 are kinda hard to tell apart, would this be better as just checking against either 

A) `0` vs `1 << N`
B) as int/long (vs uint/ulong) just check == 0 vs \< 0.

And, also, maybe a helper for keeping the branches symmetric.

Reply:
I added the helper. I won't do `== 0` vs `< 0` because that isn't accurate. The returned value needs to be exactly `0x80000000`, any other negative value is invalid.

I also think `1 << N` is harder to read/rationalize, but I don't really have a strong preference.